cordapp-cpk2/src/main/java/net/corda/plugins/cpb2/CpbTask.java

```diff
@@ -61,18 +65,29 @@ public CpbTask() {
     @NotNull
     public AbstractCopyTask from(@NotNull Object... args) {
         return super.from(args, copySpec ->
-            copySpec.exclude(this::isCPK)
+            copySpec.exclude(this::isValidCPK)
         );
     }
 
-    private boolean isCPK(@NotNull FileTreeElement element) {
+    private boolean isValidCPK(@NotNull FileTreeElement element) {
         if (!element.getName().endsWith(CPK_FILE_SUFFIX)) {
             return false;
         }
 
-        final Path cpkPath = element.getFile().toPath();
+        Path cpkPath = element.getFile().toPath();
         try (JarInputStream cpkStream = new JarInputStream(new BufferedInputStream(Files.newInputStream(cpkPath)))) {
-            final String cpkType = cpkStream.getManifest().getMainAttributes().getValue(CORDA_CPK_TYPE);
+            Manifest manifest = cpkStream.getManifest();
+            String cpkType = manifest.getMainAttributes().getValue(CORDA_CPK_TYPE);
+            String cpkCordappName = manifest.getMainAttributes().getValue(CPK_CORDAPP_NAME);
+            if (cpkCordappName != null) {
+                if (cpkNames.containsKey(cpkCordappName)) {
+                    if (!cpkNames.get(cpkCordappName).equals(manifest)) {
+                        throw new InvalidUserDataException("Two CPKs may not share a cordappCpkName. Error in " + cpkCordappName);
+                    }
+                } else {
+                    cpkNames.put(cpkCordappName, manifest);
+                }
+            }
```

Comment:
Perhaps we could try and extract this logic into some function named `checkCordappNameUniqueness`? This function is about checking if it's CPK or even valid CPK. But this is about 2 CPKs sharing the same cordapp name.